app/services/topbar_sweeper.rb

```diff
@@ -0,0 +1,22 @@
+class TopbarSweeper
+  include Singleton
+  attr_accessor :controller
+
+  def expire_cache(user = User.current)
+    controller.expire_fragment(self.class.fragment_name(user.id)) if controller.present? && user.present?
+  end
+
+  class << self
+    delegate :expire_cache, to: :instance
+
+    def fragment_name(id = User.current.id)
+      "tabs_and_title_records-#{id}"
+    end
+
+    def expire_cache_all_users
+      User.unscoped.pluck(:id).each do |id|
+        Rails.cache.delete("views/#{fragment_name(id)}")
+      end
+    end
+  end
+end
```

Comment:
I know you are just getting rid of dependency, but this class is very clunky, it was introduced as part of refactoring in issue 2513 and it's thread unfriendly singleton with unnecessary reference to controller. Method `expire_fragment` which is only available in controller context should be really called from proper place and not from here. Also the mixture of class methods and singleton methods is not ideal. I think it can be all done properly in a controller helper.